jetstream/stream_config.go

```diff
@@ -26,154 +26,356 @@ import (
+		// the API. Defaults to false.
+		DenyPurge bool `json:"deny_purge,omitempty"`
+
+		// AllowRollup allows the use of the Nats-Rollup header to replace all
+		// contents of a stream, or subject in a stream, with a single new
+		// message.
+		AllowRollup bool `json:"allow_rollup_hdrs,omitempty"`
+
+		// Compression specifies the message storage compression algorithm.
+		// Defaults to NoCompression.
+		Compression StoreCompression `json:"compression"`
+
+		// FirstSeq is the initial sequence number of the first message in the
+		// stream.
+		FirstSeq uint64 `json:"first_seq,omitempty"`
+
+		// SubjectTransform allows applying a transformation to matching
+		// messages' subjects.
 		SubjectTransform *SubjectTransformConfig `json:"subject_transform,omitempty"`
 
-		// Allow republish of the message after being sequenced and stored.
+		// RePublish allows immediate republishing a message to the configured
+		// subject after it's stored.
 		RePublish *RePublish `json:"republish,omitempty"`
 
-		// Allow higher performance, direct access to get individual messages. E.g. KeyValue
+		// AllowDirect enables direct access to individual messages using direct
+		// get API. Defaults to false.
 		AllowDirect bool `json:"allow_direct"`
-		// Allow higher performance and unified direct access for mirrors as well.
+
+		// MirrorDirect enables direct access to individual messages from the
+		// origin stream using direct get API. Defaults to false.
 		MirrorDirect bool `json:"mirror_direct"`
 
-		// Limits for consumers on this stream.
+		// ConsumerLimits defines limits of certain values that consumers can
+		// set, defaults for those who don't set these settings
 		ConsumerLimits StreamConsumerLimits `json:"consumer_limits,omitempty"`
 
-		// Metadata is additional metadata for the Stream.
-		// Keys starting with `_nats` are reserved.
-		// NOTE: Metadata requires nats-server v2.10.0+
+		// Metadata is a set of application-defined key-value pairs for
+		// associating metadata on the stream. This feature requires nats-server
+		// v2.10.0 or later.
 		Metadata map[string]string `json:"metadata,omitempty"`
+
+		// Template identifies the template that manages the Stream. DEPRECATED:
+		// This feature is no longer supported.
+		Template string `json:"template_owner,omitempty"`
 	}
 
-	// StreamSourceInfo shows information about an upstream stream source.
+	// StreamSourceInfo shows information about an upstream stream
+	// source/mirror.
 	StreamSourceInfo struct {
-		Name              string                   `json:"name"`
-		Lag               uint64                   `json:"lag"`
-		Active            time.Duration            `json:"active"`
-		FilterSubject     string                   `json:"filter_subject,omitempty"`
+		// Name is the name of the stream that is being replicated.
+		Name string `json:"name"`
+
+		// Lag informs how many messages behind the source/mirror operation is.
```

Comment:
It will only report correct lag if its actually able to communicate with the source/mirror

Reply:
Changed to:
```go
		// Lag informs how many messages behind the source/mirror operation is.
		// This will only show correctly if there is active communication
		// with stream/mirror.
```